### ISSUE TITLE
Validate schema line number

### ISSUE THREAD

**Issue by Christophvh** [nuwave/lighthouse#1189]
**What problem does this feature proposal attempt to solve?**

Speed up development workflow

**Which possible solutions should be considered?**

Currently the command: `php artisan ligthouse:validate-schema` is nice, but it would be very usefull if we got an indication on which line or in which file the schema fails.

Example of cryptic error, which is hard to find:
<img width="898" alt="Screenshot 2020-02-13 at 18 00 59" src="https://user-images.githubusercontent.com/9588070/74458992-d721ee80-4e8a-11ea-9cfd-d58ffa1a9758.png">

**Comment by spawnia** [comment]
I understand how this would be nice to have, i thought about this too. However, there are a few issues that prevent us from implementing something like this.

First of all, a schema does not necessarily come from a file. While Lighthouse reads them from disk by default, a user could implement a custom `SchemaSourceProvider` and get a schema from anywhere.

Next issue, the `SchemaSourceProvider` just returns a single string with the merged schema, having all the imports resolved. From that, we can not give accurate line numbers or any other information.

We might try to parse the files individually, but that might run into another problem: It is allowed to use `#import` in weird places, which would result in invalid schema's before they are resolved.

Finally, Lighthouse allows manipulation of the schema before validation. Many directives change the schema structurally.

In conclusion, there are a lot of design choices currently embedded deeply within Lighthouse, spread across multiple interfaces and components, that make it extremely hard to localize validation errors with any kind of accuracy.

I guess the best we could do is going through the source files one by one and checking for syntax errors. That could happen in addition to the full schema validation. Basically, we can use something similar to https://github.com/nuwave/lighthouse/blob/master/src/Schema/Source/SchemaStitcher.php to gather all files - keeping them separate - and feed them into the parser.

Feel free to give it a shot!

**Comment by Christophvh** [comment]
Thanks for the answer, yes I realize this is a difficult one. I will think about it, maybe add a PR, but like you said, not that easy.

**Comment by spawnia** [comment]
Closing this issue, as i don't see a great path forward to implement it. I am open for PRs that improve the development workflow.

**Comment by webard** [comment]
Good idea will be showing -10/+10 lines of schema where error occur. Will be a lot easier to determine where is problem. This should be possible.

**Comment by spawnia** [comment]
@webard thank you for your suggestion, please provide feedback on #1892 and potentially improve upon it.
